### DISCUSSION TITLE
netbox update from 3.5.9 to 3.6.1

### DISCUSSION THREAD

**Discussion by rottenteeth** [netbox-community/netbox#13835] [Help Wanted!]
Hi, I have trouble a problem with update Netbox from 3.5.9 to 3.6.1.

Successfully installed Pillow-9.5.0 django-auth-ldap-4.5.0 netbox_qrcode-0.0.10 nextbox-ui-plugin-0.13.0 pyasn1-modules-0.3.0 pypng-0.20220715.0 python-ldap-3.4.3 qrcode-7.4.2
Applying database migrations (python3 netbox/manage.py migrate)...
Operations to perform:
  Apply all migrations: admin, auth, circuits, contenttypes, core, dcim, django_rq, extras, ipam, nextbox_ui_plugin, sessions, social_django, taggit, tenancy, users, virtualization, wireless
Running migrations:
  No migrations to apply.
  Your models in app(s): 'nextbox_ui_plugin' have changes that are not yet reflected in a migration, and so won't be applied.
  Run 'manage.py makemigrations' to make new migrations, and then re-run 'manage.py migrate' to apply them.

**Comment by bogdancordos** [comment]
nextbox_ui_plugin is not currently supported on 3.6. Disable the plugin and try the update again.

**Comment by rottenteeth** [comment]
I disabled the plugin and tried to update.

apt update && apt upgrade
cd /opt/netbox
sudo git checkout master
sudo git pull origin master

Without any errors after ./upgrade.sh and sudo systemctl restart netbox netbox-rq I still have version 3.5.9.

**Reply by candlerb** [reply]
What responses did you get from those "git checkout" and "git pull" commands?

What do "git status" and "git show" show now?

**Comment by rottenteeth** [comment]
git checkout master

M       netbox/netbox/settings.py
Already on 'master'
Your branch is behind 'origin/master' by 140 commits, and can be fast-forwarded.
(use "git pull" to update your local branch)

git pull origin master

From https://github.com/digitalocean/netbox
* branch                master     -> FETCH_HEAD
Updating 0dbfbf694..99ab054ea
error: Your local changes to the following files would be overwritten by merge:
netbox/netbox/settings.py
Please commit your changes or stash them before you merge.
Aborting

git status

On branch master
Your branch is behind 'origin/master' by 140 commits, and can be fast-forwarded.
(use "git pull" to update your local branch)

Changes not staged for commit:
(use "git add <file>..." to update what will be committed)
(use "git restore <file>..." to discard changes in working directory)
modified:   netbox/netbox/settings.py

Untracked files:
(use "git add <file>..." to include in what will be committed)
gunicorn_config.py
netbox-3.5.0
netbox/netbox/conf.old
netbox/netbox/configuration.py.save

no changes added to commit (use "git add" and/or "git commit -a")

git show

commit 0dbfbf6941d580b19948c06aecc33161d5575a92 (HEAD -> master)
Merge: 8555269f7 d51553027
Author: Jeremy Stretch <jstretch@netboxlabs.com>
Date:   Mon Aug 28 17:07:15 2023 -0400

Merge pull request #13591 from netbox-community/develop

Correct version number


Would be this the problem?

Applying database migrations (python3 netbox/manage.py migrate)...
Operations to perform:
Apply all migrations: admin, auth, circuits, contenttypes, core, dcim, django_rq, extras, ipam, sessions, social_django, taggit, tenancy, users, virtualization, wireless
Running migrations:
No migrations to apply.

**Comment by rottenteeth** [comment]
Applying database migrations (python3 netbox/manage.py migrate)...
Operations to perform:
  Apply all migrations: admin, auth, circuits, contenttypes, core, dcim, django_rq, extras, ipam, sessions, social_django, taggit, tenancy, users, virtualization, wireless
Running migrations:
  No migrations to apply.
Checking for missing cable paths (python3 netbox/manage.py trace_paths --no-input)...
Found no missing console port paths; skipping
Found no missing console server port paths; skipping
Found no missing interface paths; skipping
Found no missing power feed paths; skipping
Found no missing power outlet paths; skipping
Found no missing power port paths; skipping
Finished.

...

Completed. Total entries: 7108
Removing expired user sessions (python3 netbox/manage.py clearsessions)...
Clearing the cache (python3 netbox/manage.py clearcache)...
Cache has been cleared.
Upgrade complete! Don't forget to restart the NetBox services:
  > sudo systemctl restart netbox netbox-rq

**Comment by candlerb** [comment]
> error: Your local changes to the following files would be overwritten by merge:
> netbox/netbox/settings.py

This means it was unable to bring the local code repository up to date (and it remains 140 commits behind the upstream repository) because you made a local change to this file. This is an internal part of Netbox that you're not supposed to change.

`git diff` will show you how you changed it. If you can show this, and explain what problem you were trying to solve, we might be able to suggest a better way.

`git checkout netbox/netbox/settings.py` will throw away your changes. Then `git pull` will work.

Alternatively:

```
git stash
git pull
git stash apply
```

will temporarily revert your local changes, update the code, and then reapply the changes. This *normally* works, and will leave your settings.py file in its modified state. However if you've modified it in a way which is incompatible with an upstream change to this file, then there will be a merge conflict that you'll have to fix (the file will have both changes, marked with `>>>>>` and `<<<<<` internally)

**Reply by candlerb** [reply]
> From https://github.com/digitalocean/netbox

FYI, the home is now https://github.com/netbox-community/netbox although there's a redirect in place.

You can edit this in `/opt/netbox/.git/config`

**Comment by rottenteeth** [comment]
git diff

diff --git a/netbox/netbox/settings.py b/netbox/netbox/settings.py
index 75099a029..171c95b65 100644
--- a/netbox/netbox/settings.py
+++ b/netbox/netbox/settings.py
@@ -94,7 +94,7 @@ CORS_ORIGIN_REGEX_WHITELIST = getattr(configuration, 'CORS_ORIGIN_REGEX_WHITELIS
 CORS_ORIGIN_WHITELIST = getattr(configuration, 'CORS_ORIGIN_WHITELIST', [])
 CSRF_COOKIE_NAME = getattr(configuration, 'CSRF_COOKIE_NAME', 'csrftoken')
 CSRF_COOKIE_SECURE = getattr(configuration, 'CSRF_COOKIE_SECURE', False)
-CSRF_TRUSTED_ORIGINS = getattr(configuration, 'CSRF_TRUSTED_ORIGINS', [])
+CSRF_TRUSTED_ORIGINS = getattr(configuration, 'CSRF_TRUSTED_ORIGINS', ['https://*.mydomain.com','http://IP'])
 DATE_FORMAT = getattr(configuration, 'DATE_FORMAT', 'N j, Y')
 DATETIME_FORMAT = getattr(configuration, 'DATETIME_FORMAT', 'N j, Y g:i a')
 DEBUG = getattr(configuration, 'DEBUG', False)

I used it and after that ./upgrade.sh worked. Migrations finished successfully.
git stash
git pull
git stash apply

**Reply by candlerb** [reply]
I see. There is no need to set CSRF_TRUSTED_ORIGINS there; set it in /opt/netbox/netbox/netbox/configuration.py and you are good.

```
# add to configuration.py
CSRF_TRUSTED_ORIGINS=['https://*.mydomain.com','http://IP']
```

Once you've done that, you can revert settings.py to the original, unmodified version.

(The code you've changed in settings.py says "pick up the value from configuration.py, and if it's not present there, use this value as a default")

**Comment by rottenteeth** [comment]
Thank you, your advice are very helpful.
